lib/rack/cache/redis_entitystore.rb

```diff
@@ -8,21 +7,26 @@ class RedisBase < self
         # The underlying ::Redis instance used to communicate with the Redis daemon.
         attr_reader :cache
 
+        class << self
+          attr_accessor :default_ttl
+        end
+
         extend Rack::Utils
 
         def open(key)
           data = read(key)
           data && [data]
         end
 
-        def self.resolve(uri)
-          new ::Redis::Store::Factory.resolve(uri.to_s)
+        def self.resolve(uri, options = {})
+          new ::Redis::Store::Factory.resolve(uri.to_s), options
         end
       end
 
       class Redis < RedisBase
         def initialize(server, options = {})
           @cache = ::Redis::Store::Factory.create(server)
+          self.class.default_ttl = options[:default_ttl] || 86_400 * 365 # 1 year
```

Comment:
As stated above, this looks like it really wants to be an instance variable. Also, you should still default this to `DEFAULT_TTL` if nothing has been given...

```ruby
@default_ttl = options[:default_ttl] || ::Redis::Rack::Cache::DEFAULT_TTL
```